tools/waffle-example/README.md

```diff
@@ -0,0 +1,72 @@
+```
+OPERATOR_PRIVATE_KEY=
+RELAY_ENDPOINT=
+```
+
+ - `OPERATOR_PRIVATE_KEY` is your account ECDSA hex-encoded private key.
+ - `RELAY_ENDPOINT` is a path to your JSON RPC Api. `https://testnet.hashio.io/api` for testnet.
+
+## Prerequisites
+
+This repository already contains the artifcats of the test Smart Contract.
+In order to recreate them on your own you can run script:
+```shell
+npm run build
+```
+It uses dockerized compiler so make sure to start the [Docker](https://www.google.com/search?client=safari&rls=en&q=Docker&ie=UTF-8&oe=UTF-8) first.
+
+## Setup & Install
+
+In the project directory:
+1. Run `npm install`
+2. Run `npm run test`
+
+# Waffle
+[Waffle](https://ethereum-waffle.readthedocs.io/) is a library designed for testing Smart Contracts.
+
+#### Features
+
+Waffle provides several features that enhance the testing experience for developers:
+
+- Includes new Chai matchers specifically designed for Smart Contract testing.
+- Enables the use of mocked wallets and Smart Contracts for isolated and controlled testing environments.
+- Allows the use of fixtures to define reusable setups for tests, reducing boilerplate code and simplifying complex test setups.
+
+#### Dependencies
+
+Waffle uses **ethers.js** to perform operations such as deploying Smart Contracts and sending transactions. It leverages **Mocha** and **Chai** as the foundational testing tools, while adding its own custom matchers tailored for Smart Contracts.
+
+## Installation:
+To install Waffle, you need to add the `ethereum-waffle` node module to your project. You can do this using the following command:
+
+```shell
+npm install --save-dev ethereum-waffle
+```
+
+## Hedera Smart Contracts Development
+
+Waffle utilizes `ethers.js` to communicate with the JSON-RPC API.
+
+A sample usage is documented in this [example](https://github.com/hashgraph/hedera-json-rpc-relay/tree/main/tools/hardhat-example).
+
+Both [contract deployment](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/tools/hardhat-example/scripts/deployContract.js) and [calls](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/tools/hardhat-example/scripts/contractCall.js) are performed using `ethers.js` in the same way Waffle does.
+
+The [Chai matchers](https://ethereum-waffle.readthedocs.io/en/latest/matchers.html) provided by Waffle are not dependent on any specific network. They can be used to easily detect common scenarios in Smart Contract testing, such as whether an event was emitted by the Smart Contract or if an operation was reverted.
+
+There are also matchers for Chai that test raw strings such as addresses and keys. These matchers assume that only ECDSA keys are correct.
+
+However, the mocks provided by Waffle are designed to simulate Ethereum network behavior, which means they may not replicate some Hedera-specific traits, such as `ecrecover` behavior for non-ECDSA keys or after private key changes. The wallet mock will only support wallets utilizing ECDSA keys.
```

Comment:
Do we have an official story or explanation for the user if they would want to use `ecrecover` for non-ECDSA keys?  Is that a realistic scenario?

Reply:
@ebadiere 

The description of this task provides an overview of what developers should expect when using ecrecover in Hedera, based on the key type they choose:
https://github.com/hashgraph/hedera-smart-contracts/issues/814#issue-2366064991

Here is a PR with a sample usage:
https://github.com/hashgraph/hedera-smart-contracts/pull/815

As far as I know, we are investigating the possibility of providing developers with an alternative to the ecrecover function that would work for non-ECDSA keys. However, work on this issue has either not started or is on your side.

Let me know if there is anything we should change in this PR. Should I include the link to the issue mentioned above in the README file?